### PR TITLE
Add Battle.net auth

### PR DESCRIPTION
**Pull request by gvlasov**
Adds Battle.net login support.

[Battle.net OAuth2 doc](https://develop.battle.net/documentation/guides/using-oauth)

Made it work with my fork of https://github.com/umputun/remark42, screenshot:

### CODE REVIEW

File: provider/providers.go [hunk 1]
```diff
@@ -147,3 +148,25 @@ func NewTwitter(p Params) Oauth1Handler {
...
+			AuthURL:   "https://eu.battle.net/oauth/authorize",
+			TokenURL:  "https://eu.battle.net/oauth/token",
+			AuthStyle: oauth2.AuthStyleInParams,
+		},
+		scopes:  []string{},
+		infoURL: "https://eu.battle.net/oauth/userinfo",
+		mapUser: func(data UserData, _ []byte) token.User {
+			userInfo := token.User{
+				ID:   "battlenet_" + token.HashID(sha1.New(), data.Value("id")),
+				Name: data.Value("battletag"),
```
Comment (umputun) [inline provider/providers.go:166]: is the lack of pictrue mapping intentional? i.e. is it supported by battlenet? If it is - we should map it

File: README.md [hunk 2]
```diff
@@ -364,6 +364,13 @@ _instructions for google oauth2 setup borrowed from [oauth2_proxy](https://githu
 
 For more details refer to [Yandex OAuth](https://tech.yandex.com/oauth/doc/dg/concepts/about-docpage/) and [Yandex.Passport](https://tech.yandex.com/passport/doc/dg/index-docpage/) API documentation.
 
+##### Battle.net Auth Provider
```
Comment (umputun) [inline README.md:367]: Will be nice to have the actcual inststuction, like other providers have

File: provider/providers.go [hunk 3]
```diff
@@ -147,3 +148,25 @@ func NewTwitter(p Params) Oauth1Handler {
...
+// NewBattlenet makes Battle.net oauth2 provider
+func NewBattlenet(p Params) Oauth2Handler {
+	return initOauth2Handler(p, Oauth2Handler{
+		name: "battlenet",
+		endpoint: oauth2.Endpoint{
+			AuthURL:   "https://eu.battle.net/oauth/authorize",
+			TokenURL:  "https://eu.battle.net/oauth/token",
+			AuthStyle: oauth2.AuthStyleInParams,
+		},
+		scopes:  []string{},
```
Comment (umputun) [inline provider/providers.go:161]: does it support scopes? if so, we should set a minimal one sufficient for the basic info we get

**Comment by umputun** [comment]
https://github.com/go-pkgz/auth/pull/53/checks?check_run_id=774369056#step:5:131

File: provider/providers.go (hunk 1 above)
Comment (gvlasov) [reply provider/providers.go:166]: @umputun There is no user picture in Battle.net, only pictures per user per game, some of those games don't even have an official API

File: README.md (hunk 2 above)
Comment (gvlasov) [reply README.md:367]: @umputun Added it in a new commit

**Comment by gvlasov** [comment]
@umputun Test error fixed

**Pull request merged**